src/main/java/org/commcare/formplayer/session/FormSession.java

```diff
@@ -395,6 +395,7 @@ private void deserializeFormDef(String serializedFormDef) throws IOException, De
 
     public SerializableFormSession serialize() throws IOException {
         session.incrementSequence();
+        session.setFormXml(serializeFormDef(formDef));
```

Comment:
@ctsims Removing this was a breaking change - it seems odd that we have to save the `formDef` every time and not just the `instanceXml`

Reply:
Definitely agree with that, it shouldn't be the case that there's any state in `formDef` which changes in between saves. 

I wonder whether it might be the case that there's some piece of initialization which the system is relying incorrectly on persisting (rather than rebuilding) between requests.

Is there a specific test I could run which demonstrates the issue? I could probably provide some insight on why it's breaking.

Reply:
This was one of the tests that broke: [CaseTests.testCaseCreate](https://github.com/dimagi/formplayer/blob/a1e599955b89bea8a95cf16b98ee35c21a91323c/src/test/java/org/commcare/formplayer/tests/CaseTests.java#L32). Submitting the form failed because the case ID in the form was not set.

Reply:
bb02e1bd